src/lib/core/datetime/calendar-locale.ts

```diff
@@ -0,0 +1,168 @@
+   */
+  getCalendarYearHeaderLabel: (date: SimpleDate) => string;
+}
+
+
+/**
+ * The default implementation of CalendarLocale. This implementation is a best attempt at
+ * localization using only the functionality natively available in JS. If more robust localization
+ * is needed, an alternate class can be provided as the CalendarLocale for the app.
+ */
+export class DefaultCalendarLocale implements  CalendarLocale {
+  months = SUPPORTS_INTL_API ?  this._createMonthsArray('long') :
+      [
+        'January',
+        'February',
+        'March',
+        'April',
+        'May',
+        'June',
+        'July',
+        'August',
+        'September',
+        'October',
+        'November',
+        'December'
+      ];
+
+  shortMonths = SUPPORTS_INTL_API ? this._createMonthsArray('short') :
+      ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
+
+  narrowMonths = SUPPORTS_INTL_API ? this._createMonthsArray('narrow') :
+      ['J', 'F', 'M', 'A', 'M', 'J', 'J', 'A', 'S', 'O', 'N', 'D'];
+
+  days = SUPPORTS_INTL_API ? this._createDaysArray('long') :
+      ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+
+  shortDays = SUPPORTS_INTL_API ? this._createDaysArray('short') :
+      ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
+
+  narrowDays = SUPPORTS_INTL_API ? this._createDaysArray('narrow') :
+      ['S', 'M', 'T', 'W', 'T', 'F', 'S'];
+
+  dates = [null].concat(
+      SUPPORTS_INTL_API ? this._createDatesArray('numeric') :  range(31, i => String(i + 1)));
+
+  firstDayOfWeek = 0;
+
+  calendarLabel = 'Calendar';
+
+  openCalendarLabel = 'Open calendar';
+
+  parseDate(value: any) {
+    if (value instanceof SimpleDate) {
+      return value;
+    }
+    let timestamp = typeof value == 'number' ? value : Date.parse(value);
+    return isNaN(timestamp) ? null : SimpleDate.fromNativeDate(new Date(timestamp));
+  }
+
+  formatDate = this._createFormatFunction(
+      undefined, (date: SimpleDate) => date.toNativeDate().toDateString());
+
+  getCalendarMonthHeaderLabel = this._createFormatFunction(
+      {month: 'short', year: 'numeric'},
+      (date: SimpleDate) => this.shortMonths[date.month] + ' ' + date.year);
+
+  getCalendarYearHeaderLabel = this._createFormatFunction(
+      {year: 'numeric'}, (date: SimpleDate) => String(date.year));
+
+  private _createMonthsArray(format: string) {
+    let dtf = new Intl.DateTimeFormat(undefined, {month: format});
+    return range(12, i => dtf.format(new Date(2017, i, 1)));
+  }
+
+  private _createDaysArray(format: string) {
+    let dtf = new Intl.DateTimeFormat(undefined, {weekday: format});
+    return range(7, i => dtf.format(new Date(2017, 0, i + 1)));
+  }
+
+  private _createDatesArray(format: string) {
+    let dtf = new Intl.DateTimeFormat(undefined, {day: format});
+    return range(31, i => dtf.format(new Date(2017, 0, i + 1)));
+  }
+
+  private _createFormatFunction(
```

Comment:
done, also simplified it since its just as easy to return null and specify the fallback elsewhere